cns/cniconflist/generator_linux_test.go

```diff
@@ -0,0 +1,77 @@
+package cniconflist_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/cni"
+	"github.com/Azure/azure-container-networking/cns/cniconflist"
+	"github.com/Azure/azure-container-networking/cns/configuration"
+	"github.com/stretchr/testify/assert"
+)
+
+type bufferWriteCloser struct {
+	*bytes.Buffer
+}
+
+func (b *bufferWriteCloser) Close() error {
+	return nil
+}
+
+func TestGenerateConflist(t *testing.T) {
+	generatorTests := []struct {
+		s                       configuration.CNIConflistScenario
+		fixture                 string
+		expectedNewGeneratorErr bool
+	}{
+		{
+			s:       configuration.CNIConflistScenarioV4Overlay,
+			fixture: "testdata/fixtures/azure-linux-swift-overlay.conflist",
+		},
+		{
+			s:                       configuration.CNIConflistScenario("unknown"),
+			expectedNewGeneratorErr: true,
+		},
+	}
+
+	for _, test := range generatorTests {
+		tt := test
+		t.Run(fmt.Sprintf("generate-%s", tt.s), func(t *testing.T) {
+			buffer := new(bytes.Buffer)
+			g, err := cniconflist.NewGenerator(&bufferWriteCloser{buffer}, tt.s)
+			if err != nil {
+				if tt.expectedNewGeneratorErr {
+					return
+				}
+				t.Fail()
```

Comment:
I'd combine these, just for readability, and editorialize the error a little:

```go
if err != nil && !tt.expectedNewGeneratorErr {
       t.Fatal("unexpected error creating generator: err:", err)
}

if err == nil && tt.expectedNewGeneratorErr {
       t.Fatal("exepcted an error creating generator but received none")
}
```

This reads, "if the error was present and we didn't expect one, fail" and "if there was no error, but we expected one, fail". `t.Fatal`, I assume, (equivalent of `t.Log` and `t.FailNow`) because it's nonsensical to make any assertions after that point in either circumstance.

Reply:
agreed, changed to the suggestion